lib/active_merchant/billing/gateways/pay_gate_xml.rb

```diff
@@ -251,6 +270,13 @@ def commit(action, request)
           :authorization  => response[:tid]
         )
       end
+      def commit_capture(action, authorization, request)
```

Comment:
I'd suggest you look through a mature integration like https://github.com/Shopify/active_merchant/blob/master/lib/active_merchant/billing/gateways/stripe.rb for ideas on how to structure your code to cleanly handle many cases.